Sources/Extraction/Filters/HillOrientation.c

```diff
@@ -1,26 +1,183 @@
 #include <stdlib.h>
 #include <assert.h>
+#include <math.h>
+#include <stdio.h>
 
 #include "HillOrientation.h"
+#include "General/Array.h"
 
-HillOrientation HillOrientation_Construct(void)
+#define MIN(X, Y) (((X) < (Y)) ? (X) : (Y))
+#define MAX(X, Y) (((X) > (Y)) ? (X) : (Y))
+#define NUM_VECTORS (28)
+
+const Point neighbors[NUM_VECTORS] = {
+    { -1, 1, }, { -1, 2 }, { -1, 3 },
+    { -2, 1  }, { -2, 2 }, { -2, 3 },
+    { -3, 1  }, { -3, 2 }, { -3, 3 },
+    { 0, 1 }, { 0, 2, }, { 0, 3 },
+    { 1, 0 }, { 1, 1, }, { 1, 2 }, { 1, 3 },
+    { 2, 0 }, { 2, 1  }, { 2, 2 }, { 2, 3 },
+    { 3, 0 }, { 3, 1  }, { 3, 2 }, { 3, 3 },
+    { -5, 0 }, { -5, 5 }, { 0, 5 }, { 5, 5 }
+};
+
```

Comment:
Perosnally I'd have done const Point neighbors[]
folloowed by 
#define NUM_VECTORS  sizeof(neighbors)/sizeof(neighbors[0]) 
and then you can add to these as you like